src/SourceBuild/tarball/content/eng/pipelines/vmr-ci.yml

```diff
@@ -0,0 +1,51 @@
+# This is the main build definition (PR+CI) for dotnet/dotnet
+
+trigger:
+  batch: true
+  branches:
+    include:
+    - main
+    - release/*
+    - internal/release/*
+
+pr:
+  branches:
+    include:
+    - main
+    - release/*
+    - internal/release/*
+
+stages:
+# For rolling builds we want to build the MSFT SDK first so that we can
+# compare the contents with the source-built one later.
+# This only works because we don't run this test in PRs. If we decided
+# to run this test in PRs, we would need to build the MSFT SDK there too.
+- ${{ if and(ne(variables['System.TeamProject'], 'public'), notin(variables['Build.Reason'], 'PullRequest')) }}:
+  - stage: Build_MSFT_SDK
+    displayName: Build MSFT SDK
+    jobs:
+    - template: /src/installer/eng/build.yml
+      parameters:
+        agentOs: Linux
+        jobName: Build_LinuxPortable_Release_x64
+        buildConfiguration: Release
+        buildArchitecture: x64
+        linuxPortable: true
+        runTests: false
+        isBuiltFromVmr: true
+    - template: /src/installer/eng/build.yml
+      parameters:
+        agentOs: Linux
+        jobName: Build_Arm64_Release
+        buildConfiguration: Release
+        buildArchitecture: arm64
+        runtimeIdentifier: 'linux-arm64'
+        linuxPortable: true
+        runTests: false
+        isBuiltFromVmr: true
+
+- template: /src/installer/eng/pipelines/templates/stages/vmr-build.yml
+  parameters:
+    isBuiltFromVmr: true
+    ${{ if and(ne(variables['System.TeamProject'], 'public'), notin(variables['Build.Reason'], 'PullRequest')) }}:
+      dependsOn: Build_MSFT_SDK
```

Comment:
I have some reservations about this dependency.  Pre-vmr, if there was a build break in installer, we still had the ability to run the tarball build, the tests would fail because of a missing MSFT SDK tarball but that was minor.  With this change, source-build validation would be blocked.

This ability is important during monthly servicing where breaks can happen and the schedule is tight.

Reply:
Maybe if we have a variable that could be set to disable the sdk tests would be sufficient.

Reply:
@MichaelSimons seems like there is `condition: always()` for stages as well. I guess that works?